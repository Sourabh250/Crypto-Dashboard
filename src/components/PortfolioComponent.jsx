import React from "react";
import PieChart from "../charts/PieChart";

function PortfolioComponent() {
    const data = {
        labels: ["Ethereum","Tether","Luna"],
        datasets: [
            {
                label: "Holdings",
                data: [250,375,375],
                borderWidth: 0,
            }
        ]
    };

    // Total value of holdings
    const total = data.datasets[0].data.reduce((acc,val) => acc + val,0);

    return (
        <div className="bg-gray-100 rounded-lg shadow-lg py-4 px-8 flex flex-col">
            <div className="flex items-center justify-between gap-4">
            <h1 className="text-center text-xl font-semibold">Portfolio</h1>
            <h1 className="font-light">Total value: <span className="font-semibold">${total}</span></h1>
            </div>

            {/* Pie chart */}
            <div className="flex justify-center items-center mt-4">
                <PieChart ChartData={data} />
            </div>
        </div>
    )
}

export default PortfolioComponent;