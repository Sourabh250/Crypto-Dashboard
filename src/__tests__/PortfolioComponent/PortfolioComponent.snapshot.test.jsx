import React from "react";
import renderer from 'react-test-renderer';
import PortfolioComponent from '../../components/PortfolioComponent';

test('PortfolioComponent snapshot', () => {
    const tree = renderer.create(<PortfolioComponent />).toJSON();

    expect(tree).toMatchSnapshot();
})