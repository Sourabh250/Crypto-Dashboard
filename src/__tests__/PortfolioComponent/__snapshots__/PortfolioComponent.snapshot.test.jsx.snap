// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortfolioComponent snapshot 1`] = `
<div
  className="bg-gray-100 rounded-lg shadow-lg py-4 px-8 flex flex-col"
>
  <div
    className="flex items-center justify-between gap-4"
  >
    <h1
      className="text-center text-xl font-semibold"
    >
      Portfolio
    </h1>
    <h1
      className="font-light"
    >
      Total value: 
      <span
        className="font-semibold"
      >
        $
        1000
      </span>
    </h1>
  </div>
  <div
    className="flex justify-center items-center mt-4"
  >
    <canvas
      height={150}
      role="img"
      width={300}
    />
  </div>
</div>
`;
